File: app/src/main/java/edu/gatech/cs2340/spacetrader/entity/ResourceLevel.java
package edu.gatech.cs2340.spacetrader.entity;

import java.io.Serializable;

public enum ResourceLevel implements Serializable {
    NONE("NOSPECIALRESOURCES"),
    M_RICH("MINERALRICH"),
    M_POOR("MINERALPOOR"),
    DESERT("DESERT"),
    WATER("LOTSOFWATER"),
    S_RICH("RICHSOIL"),
    S_POOR("POORSOIL"),
    F_RICH("RICHFAUNA"),
    LIFELESS("LIFELESS"),
    SHROOMS("WEIRDMUSHROOMS"),
    HERBS("LOTSOFHERBS"),
    ART("ARTISTIC"),
    WAR("WARLIKE");


    /** the full string representation of the difficulty */
    private final String name;

    /**
     * Constructor for the enumeration
     *
     * @param name   full name of the techLevel
     */
    ResourceLevel(String name) {
        this.name = name;
    }

    /**
     *
     * @return the display string representation of the difficulty
     */
    public String toString() { return name; }
}

File: Extra Credit/Python/entity/resourcelevel.py
import enum
import random


class ResourceLevel(enum.Enum):
    NONE = "NOSPECIALRESOURCES", "No Special Resources"
    M_RICH = "MINERALRICH", "Mineral Rich"
    M_POOR = "MINERALPOOR", "Mineral Poor"
    DESERT = "DESERT", "Desert"
    WATER = "LOTSOFWATER", "Lots of Water"
    S_RICH = "RICHSOIL", "Rich Soil"
    S_POOR = "POORSOIL", "Poor Soil"
    F_RICH = "RICHFAUNA", "Rich Fauna"
    LIFELESS = "LIFELESS", "Lifeless"
    SHROOMS = "WEIRDMUSHROOMS", "Weird Mushrooms"
    HERBS = "LOTSOFHERBS", "Lots of Herbs"
    ART = "ARTISTIC", "Artistic"
    WAR = "WARLIKE", "Warlike"

    def __init__(self, id, full_name):
        self.id = id
        self.full_name = full_name

    def __str__(self):
        return self.full_name

    @classmethod
    def random(cls):
        resources = []
        for resource in cls:
            resources.append(resource)
        return random.choice(resources)
